numba/core/pythonapi.py

```diff
@@ -134,7 +134,13 @@ def read_const(self, index):
         """
         assert index < len(self.env.consts)
 
-        return self.pyapi.list_getitem(self.env_body.consts, index)
+        builder = self.pyapi.builder
+        consts = self.env_body.consts
+        ret = cgutils.alloca_once(builder, self.pyapi.pyobj, zfill=True)
+        with builder.if_then(cgutils.is_not_null(builder, consts)):
```

Comment:
If `consts` is NULL, what does this imply? Has something gone wrong if something is trying to index a nullptr? If so it ought be caught?

Reply:
If it is NULL, it is probably `env` got removed. I can make it set an exception.